src/api/java/doggytalents/api/inferface/AbstractDog.java

```diff
@@ -97,4 +100,19 @@ public TranslatableComponent getGenderTip() {
     public TranslatableComponent getGenderName() {
         return this.getTranslationKey(EnumGender::getUnlocalisedName);
     }
+
+    public void setNavigation(PathNavigation p) {
```

Comment:
Change to the AbstractDog class, made the Dog::navigation and Dog::moveControl writable for IDogAlterations or Goals to alter. Which made it possible to hot-swap between Swimming MoveControl and Ground MoveControl